packages/from-solana/src/crypto/SolanaDomainPublicKey.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  AccountPublicKey,
+  AuthenticationKey,
+  Ed25519PublicKey,
+  hashValues,
+  Hex,
+  Serializer,
+  VerifySignatureArgs,
+} from '@aptos-labs/ts-sdk';
+import { computeDomainAuthenticationKey } from './abstraction';
+import { makeSolanaSigningMessage } from './makeSolanaSigningMessage';
+import { SolanaDomainIdentity } from './SolanaDomainIdentity';
+
+// Should extend AbstractPublicKey, but it's currently not exported
+export class SolanaDomainPublicKey extends AccountPublicKey {
+  readonly identity: SolanaDomainIdentity;
+  readonly _authKey: AuthenticationKey;
+
+  constructor(identity: SolanaDomainIdentity, authenticationFunction: string) {
+    super();
+    this.identity = identity;
+
+    const utf8EncodedDomain = new TextEncoder().encode(this.identity.domain);
+    const solanaPublicKeyBytes = this.identity.solanaPublicKey.toBytes();
+
+    const serializer = new Serializer();
+    serializer.serializeBytes(utf8EncodedDomain);
+    serializer.serializeFixedBytes(solanaPublicKeyBytes); // fixed length 32 bytes
+    const accountIdentifier = hashValues([serializer.toUint8Array()]);
+
+    this._authKey = computeDomainAuthenticationKey(
+      authenticationFunction,
+      accountIdentifier,
+    );
+  }
+
+  authKey(): AuthenticationKey {
+    return this._authKey;
+  }
+
+  verifySignature({ message, signature }: VerifySignatureArgs): boolean {
+    const messageBytes = Hex.fromHexInput(message).toUint8Array();
+    const solanaSigningMessage = makeSolanaSigningMessage(this.identity, messageBytes);
+
+    // Solana uses Ed25519 just like Aptos
```

Comment:
if solana uses the same ED25519 can't we just extend/use the Ed25519PublicKey class from the ts-sdk?

Reply:
In which context? Like extending `SolanaDerivedPublicKey` from `Ed25519PublicKey`?
I'd rather go with composition rather than subclassing in this case, it wouldn't add much value extending the class.